### PR TITLE
lint(): fix eslint errors

### PR DESCRIPTION
**Pull request by robothot**
## Description

Make lint PASS!

## Changes

var => let/const
eslint ignore

### CODE REVIEW

**Comment by robothot** [comment]
@ShaMan123 Sorry, for wasting your time earlier, I closed the previous PR and now I open a new PR to merge into the master branch. Sorry for always wasting your time with simple questions.

**Comment by ShaMan123** [comment]
You are being too harsh on yourself.
Git and contributing is a learning curve. For me as well.
I did my fair share of mistakes and time sinks.
Rebasing is really simple via the PR edit button, just so you know.
Keep contributing and feel free to ask for guidance.

**Comment by ShaMan123** [comment]
take a look at the failing tests and fix what is wrong.
each time you push the tests re run.

**Comment by ShaMan123** [comment]
Start from the build test

image

> [!] (plugin babel) SyntaxError: /home/runner/work/fabric.js/fabric.js/src/parser/getGradientDefs.ts: Identifier 'j' has already been declared. (23:6)

**Comment by ShaMan123** [comment]
Then once all tests pass (lint can fail) request my review

**Comment by robothot** [comment]
@ShaMan123  ok, I  will try, but My time zone +8, I'll try to fix this tomorrow

**Comment by robothot** [comment]
@ShaMan123 Can you still execute CI? I seem to run `npm run test` locally and report an error

**Comment by ShaMan123** [comment]
seems you didn't install deps
`npm i --include-dev`

**Comment by robothot** [comment]
image

```
TAP version 13
ok 1 Clipping shapes > Clip a rect with a circle, no zoom
ok 2 Clipping shapes > A clippath ignores fill and stroke for drawing, not positioning
ok 3 Clipping shapes > falsy values for fill are handled
ok 4 Clipping shapes > Clip a rect with a circle, with zoom
ok 5 Clipping shapes > Clip a group with a circle
ok 6 Clipping shapes > ClipPath can be transformed
ok 7 Clipping shapes > ClipPath can be a group with many objects
ok 8 Clipping shapes > ClipPath can be inverted, it will clip what is outside the clipPath
ok 9 Clipping shapes > an absolute positioned clipPath, shared
ok 10 Clipping shapes > a clipPath on the canvas
ok 11 Clipping shapes > clipPath with a path on a simple elements
ok 12 Clipping shapes > clipPath made of polygons and paths
ok 13 Controls rendering > Rect with default controls
ok 14 Controls rendering > Rect with padding
ok 15 Controls rendering > transparent corner square
ok 16 Controls rendering > circle corners green
ok 17 Controls rendering > circle corners solid red
ok 18 Controls rendering > corner size 20 rect style
ok 19 Controls rendering > corner size 20 rect style
ok 20 Controls rendering > control visibility
ok 21 Controls rendering > control with fill and stroke and dash array
ok 22 Controls rendering > control with fill and stroke and dash array with borderScaleFactor
ok 23 Controls rendering > controlbox with flipped X
ok 24 Controls rendering > controlboxes with skewY
ok 25 Controls rendering > controlboxes with skewY and flipX
ok 26 Free Drawing > Simple free drawing (result)
ok 27 Free Drawing > Simple free drawing, large brush no offset (result)
ok 28 Free Drawing > Simple free drawing, with shadow (result)
ok 29 Free Drawing > Simple free drawing, with opacity (result)
ok 30 Free Drawing > Simple free drawing to dot (result)
ok 31 Free Drawing > Simple free drawing, with high decimation (result)
ok 32 Free Drawing > Pattern src from `getPatternSrc` (result)
ok 33 Free Drawing > Pattern src from `source` (result)
ok 34 Generic rendering > Rect with strokeUniform: true
ok 35 Generic rendering > Rect DropShadow with nonScaling: true
ok 36 Generic rendering > polygon position independently from strokeWidth and origin
ok 37 Generic rendering > canvas can have a gradient background
ok 38 Generic rendering > canvas can have a gradient background and being zoomed
ok 39 Generic rendering > canvas can have a gradient background with zoom but being unaffected
ok 40 Generic rendering > objects in activeSelection toCanvasElement
ok 41 Generic rendering > canvas with background pattern and export
ok 42 Generic rendering > canvas with background pattern and multiplier
ok 43 Generic rendering > fabric.Image with imageSmoothing false
ok 44 Generic rendering > gradient should be applied to path
ok 45 Gradient stroke > Use the gradient strokeStyle for line(other shape is ok)
ok 46 Text gradient fill > Use the gradient fillStyle for text
ok 47 globalCompositeOperation > color
ok 48 globalCompositeOperation > color-burn
ok 49 globalCompositeOperation > color-dodge
ok 50 globalCompositeOperation > copy
ok 51 globalCompositeOperation > darken
ok 52 globalCompositeOperation > destination-atop
ok 53 globalCompositeOperation > destination-in
ok 54 globalCompositeOperation > destination-out
ok 55 globalCompositeOperation > destination-over
ok 56 globalCompositeOperation > difference
ok 57 globalCompositeOperation > exclusion
ok 58 globalCompositeOperation > hard-light
ok 59 globalCompositeOperation > hue
ok 60 globalCompositeOperation > lighten
ok 61 globalCompositeOperation > lighter
ok 62 globalCompositeOperation > luminosity
ok 63 globalCompositeOperation > multiply
ok 64 globalCompositeOperation > overlay
ok 65 globalCompositeOperation > saturation
ok 66 globalCompositeOperation > screen
ok 67 globalCompositeOperation > soft-light
ok 68 globalCompositeOperation > source-atop
ok 69 globalCompositeOperation > source-in
ok 70 globalCompositeOperation > source-out
ok 71 globalCompositeOperation > source-over
ok 72 globalCompositeOperation > xor
ok 73 Group Layout > fit-content layout
ok 74 Group Layout > fit-content layout relative
ok 75 Group Layout > fit-content relayout
ok 76 Group Layout > fit-content layout
ok 77 Group Layout > fit-content layout
ok 78 Group Layout > unit test scene
ok 79 Group Layout > fit-content layout after change
ok 80 Group Layout > fit-content layout add object
ok 81 Group Layout > clip-path layout
ok 82 Group Layout > clip-path layout scaleX value
ok 83 Group Layout > clip-path layout left, top, originX, originY, scaleX values - WRONG
ok 84 Group Layout > absolute clip-path layout
ok 85 Group Layout > layout with originX=left, originY=top and angle=0 values
ok 86 Group Layout > layout with originX=left, originY=center and angle=0 values
ok 87 Group Layout > layout with originX=left, originY=bottom and angle=0 values
ok 88 Group Layout > layout with originX=center, originY=top and angle=0 values
ok 89 Group Layout > layout with originX=center, originY=center and angle=0 values
ok 90 Group Layout > layout with originX=center, originY=bottom and angle=0 values
ok 91 Group Layout > layout with originX=right, originY=top and angle=0 values
ok 92 Group Layout > layout with originX=right, originY=center and angle=0 values
ok 93 Group Layout > layout with originX=right, originY=bottom and angle=0 values
ok 94 Group Layout > layout with originX=left, originY=top and angle=30 values
ok 95 Group Layout > layout with originX=left, originY=center and angle=30 values
ok 96 Group Layout > layout with originX=left, originY=bottom and angle=30 values
ok 97 Group Layout > layout with originX=center, originY=top and angle=30 values
ok 98 Group Layout > layout with originX=center, originY=center and angle=30 values
ok 99 Group Layout > layout with originX=center, originY=bottom and angle=30 values
ok 100 Group Layout > layout with originX=right, originY=top and angle=30 values
ok 101 Group Layout > layout with originX=right, originY=center and angle=30 values
ok 102 Group Layout > layout with originX=right, originY=bottom and angle=30 values
ok 103 Group Layout > layout with originX=left, originY=top and angle=60 values
ok 104 Group Layout > layout with originX=left, originY=center and angle=60 values
ok 105 Group Layout > layout with originX=left, originY=bottom and angle=60 values
ok 106 Group Layout > layout with originX=center, originY=top and angle=60 values
ok 107 Group Layout > layout with originX=center, originY=center and angle=60 values
ok 108 Group Layout > layout with originX=center, originY=bottom and angle=60 values
ok 109 Group Layout > layout with originX=right, originY=top and angle=60 values
ok 110 Group Layout > layout with originX=right, originY=center and angle=60 values
ok 111 Group Layout > layout with originX=right, originY=bottom and angle=60 values
ok 112 Group Layout > layout with originX=left, originY=top and angle=90 values
ok 113 Group Layout > layout with originX=left, originY=center and angle=90 values
ok 114 Group Layout > layout with originX=left, originY=bottom and angle=90 values
ok 115 Group Layout > layout with originX=center, originY=top and angle=90 values
ok 116 Group Layout > layout with originX=center, originY=center and angle=90 values
ok 117 Group Layout > layout with originX=center, originY=bottom and angle=90 values
ok 118 Group Layout > layout with originX=right, originY=top and angle=90 values
ok 119 Group Layout > layout with originX=right, originY=center and angle=90 values
ok 120 Group Layout > layout with originX=right, originY=bottom and angle=90 values
ok 121 Group Layout > layout with originX=left, originY=top and angle=120 values
ok 122 Group Layout > layout with originX=left, originY=center and angle=120 values
ok 123 Group Layout > layout with originX=left, originY=bottom and angle=120 values
ok 124 Group Layout > layout with originX=center, originY=top and angle=120 values
ok 125 Group Layout > layout with originX=center, originY=center and angle=120 values
ok 126 Group Layout > layout with originX=center, originY=bottom and angle=120 values
ok 127 Group Layout > layout with originX=right, originY=top and angle=120 values
ok 128 Group Layout > layout with originX=right, originY=center and angle=120 values
ok 129 Group Layout > layout with originX=right, originY=bottom and angle=120 values
ok 130 Group Layout > layout with originX=left, originY=top and angle=150 values
ok 131 Group Layout > layout with originX=left, originY=center and angle=150 values
ok 132 Group Layout > layout with originX=left, originY=bottom and angle=150 values
ok 133 Group Layout > layout with originX=center, originY=top and angle=150 values
ok 134 Group Layout > layout with originX=center, originY=center and angle=150 values
ok 135 Group Layout > layout with originX=center, originY=bottom and angle=150 values
ok 136 Group Layout > layout with originX=right, originY=top and angle=150 values
ok 137 Group Layout > layout with originX=right, originY=center and angle=150 values
ok 138 Group Layout > layout with originX=right, originY=bottom and angle=150 values
ok 139 Group Layout > layout with originX=left, originY=top and angle=180 values
ok 140 Group Layout > layout with originX=left, originY=center and angle=180 values
ok 141 Group Layout > layout with originX=left, originY=bottom and angle=180 values
ok 142 Group Layout > layout with originX=center, originY=top and angle=180 values
ok 143 Group Layout > layout with originX=center, originY=center and angle=180 values
ok 144 Group Layout > layout with originX=center, originY=bottom and angle=180 values
ok 145 Group Layout > layout with originX=right, originY=top and angle=180 values
ok 146 Group Layout > layout with originX=right, originY=center and angle=180 values
ok 147 Group Layout > layout with originX=right, originY=bottom and angle=180 values
ok 148 Group Layout > layout with originX=left, originY=top and angle=210 values
ok 149 Group Layout > layout with originX=left, originY=center and angle=210 values
ok 150 Group Layout > layout with originX=left, originY=bottom and angle=210 values
ok 151 Group Layout > layout with originX=center, originY=top and angle=210 values
ok 152 Group Layout > layout with originX=center, originY=center and angle=210 values
ok 153 Group Layout > layout with originX=center, originY=bottom and angle=210 values
ok 154 Group Layout > layout with originX=right, originY=top and angle=210 values
ok 155 Group Layout > layout with originX=right, originY=center and angle=210 values
ok 156 Group Layout > layout with originX=right, originY=bottom and angle=210 values
ok 157 Group Layout > layout with originX=left, originY=top and angle=240 values
ok 158 Group Layout > layout with originX=left, originY=center and angle=240 values
ok 159 Group Layout > layout with originX=left, originY=bottom and angle=240 values
ok 160 Group Layout > layout with originX=center, originY=top and angle=240 values
ok 161 Group Layout > layout with originX=center, originY=center and angle=240 values
ok 162 Group Layout > layout with originX=center, originY=bottom and angle=240 values
ok 163 Group Layout > layout with originX=right, originY=top and angle=240 values
ok 164 Group Layout > layout with originX=right, originY=center and angle=240 values
ok 165 Group Layout > layout with originX=right, originY=bottom and angle=240 values
ok 166 Group Layout > layout with originX=left, originY=top and angle=270 values
ok 167 Group Layout > layout with originX=left, originY=center and angle=270 values
ok 168 Group Layout > layout with originX=left, originY=bottom and angle=270 values
ok 169 Group Layout > layout with originX=center, originY=top and angle=270 values
ok 170 Group Layout > layout with originX=center, originY=center and angle=270 values
ok 171 Group Layout > layout with originX=center, originY=bottom and angle=270 values
ok 172 Group Layout > layout with originX=right, originY=top and angle=270 values
ok 173 Group Layout > layout with originX=right, originY=center and angle=270 values
ok 174 Group Layout > layout with originX=right, originY=bottom and angle=270 values
ok 175 Group Layout > layout with originX=left, originY=top and angle=300 values
ok 176 Group Layout > layout with originX=left, originY=center and angle=300 values
ok 177 Group Layout > layout with originX=left, originY=bottom and angle=300 values
ok 178 Group Layout > layout with originX=center, originY=top and angle=300 values
ok 179 Group Layout > layout with originX=center, originY=center and angle=300 values
ok 180 Group Layout > layout with originX=center, originY=bottom and angle=300 values
ok 181 Group Layout > layout with originX=right, originY=top and angle=300 values
ok 182 Group Layout > layout with originX=right, originY=center and angle=300 values
ok 183 Group Layout > layout with originX=right, originY=bottom and angle=300 values
ok 184 Group Layout > layout with originX=left, originY=top and angle=330 values
ok 185 Group Layout > layout with originX=left, originY=center and angle=330 values
ok 186 Group Layout > layout with originX=left, originY=bottom and angle=330 values
ok 187 Group Layout > layout with originX=center, originY=top and angle=330 values
ok 188 Group Layout > layout with originX=center, originY=center and angle=330 values
ok 189 Group Layout > layout with originX=center, originY=bottom and angle=330 values
ok 190 Group Layout > layout with originX=right, originY=top and angle=330 values
ok 191 Group Layout > layout with originX=right, originY=center and angle=330 values
ok 192 Group Layout > layout with originX=right, originY=bottom and angle=330 values
ok 193 Image resize filter test > Image resize with canvas zoom
ok 194 Image resize filter test > Image resize without zoom
ok 195 Image resize filter test > Image resize with scaleY != scaleX
ok 196 Image resize filter test > Image resize with scaled group
ok 197 Image Blend test > Blend image test with flip
ok 198 Image Blend test > Blend image test
ok 199 Simple svg import test > Svg import test svg_stroke_1
ok 200 Simple svg import test > Svg import test svg_stroke_2
ok 201 Simple svg import test > Svg import test svg_stroke_3
ok 202 Simple svg import test > Svg import test svg_stroke_4
ok 203 Simple svg import test > Svg import test svg_stroke_5
ok 204 Simple svg import test > Svg import test svg_stroke_6
ok 205 Simple svg import test > Svg import test svg_stroke_7
ok 206 Simple svg import test > Svg import test svg_stroke_8
ok 207 Simple svg import test > Svg import test svg_linear_1
ok 208 Simple svg import test > Svg import test svg_linear_2
ok 209 Simple svg import test > Svg import test svg_linear_3
ok 210 Simple svg import test > Svg import test svg_linear_4
ok 211 Simple svg import test > Svg import test svg_linear_5
ok 212 Simple svg import test > Svg import test svg_linear_6
ok 213 Simple svg import test > Svg import test svg_linear_7
ok 214 Simple svg import test > Svg import test svg_linear_8
ok 215 Simple svg import test > Svg import test svg_linear_9
ok 216 Simple svg import test > Svg import test svg_radial_1
ok 217 Simple svg import test > Svg import test svg_radial_2
ok 218 Simple svg import test > Svg import test svg_radial_3
ok 219 Simple svg import test > Svg import test svg_radial_4
ok 220 Simple svg import test > Svg import test svg_radial_5
ok 221 Simple svg import test > Svg import test svg_radial_6
ok 222 Simple svg import test > Svg import test svg_radial_8
ok 223 Simple svg import test > Svg import test svg_radial_9
ok 224 Simple svg import test > Svg import test svg_radial_10
ok 225 Simple svg import test > Svg import test svg_radial_11
ok 226 Simple svg import test > Svg import test svg_radial_12
ok 227 Simple svg import test > Svg import test svg_radial_13
ok 228 Simple svg import test > Svg import test svg_text_letterspacing
ok 229 Simple svg import test > Svg import test clippath-5
ok 230 Simple svg import test > Svg import test clippath-6

(process:3860): Pango-WARNING **: 13:30:37.319: couldn't load font "EnglischeSchT Not-Rotated 400px", falling back to "Sans Not-Rotated 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:37.331: couldn't load font "EnglischeSchT 400px", falling back to "Sans 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:37.343: couldn't load font "Copperplate Gothic Bold, Not-Rotated 400px", falling back to "Sans Not-Rotated 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:37.344: couldn't load font "Copperplate Gothic Bold, 400px", falling back to "Sans 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:37.395: couldn't load font "EnglischeSchT Not-Rotated 98.3095703125px", falling back to "Sans Not-Rotated 98.3095703125px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:37.408: couldn't load font "Copperplate Gothic Bold, Not-Rotated 113.9892578125px", falling back to "Sans Not-Rotated 113.9892578125px", expect ugly output.
ok 231 Simple svg import test > Svg import test clippath-7
ok 232 Simple svg import test > Svg import test clippath-9
ok 233 Simple svg import test > Svg import test vector-effect
ok 234 Simple svg import test > Svg import test svg-with-no-dim-rect
ok 235 Simple svg import test > Svg import test notoemoji-person
ok 236 Simple svg import test > Svg import test emoji-b
ok 237 Simple svg import test > Svg import test gold-logo
ok 238 Simple svg import test > Svg import test svg_missing_clippath
ok 239 Simple svg import test > Svg import test image-rendering-attr
ok 240 Simple svg import test > Svg import test svg-missing-images
ok 241 Simple svg import test > Svg import test nested-svgs
ok 242 Simple svg import test > Svg import test arc1
ok 243 Simple svg import test > Svg import test arc2
ok 244 Simple svg import test > Svg import test arc3
ok 245 Simple svg import test > Svg import test cs
ok 246 Simple svg import test > Svg import test qt
ok 247 Simple svg import test > Svg import test generic-path
ok 248 Simple svg import test > Svg import test 177
ok 249 Simple svg import test > Svg import test polygons
ok 250 Simple svg import test > Svg import test polygons-rounded
ok 251 Text > simple text test
ok 252 Text > Text with trailing spaces
ok 253 Text > Text with styles
ok 254 Text > Text on a transformed canvas

(process:3860): Pango-WARNING **: 13:30:42.684: couldn't load font "Ubuntu Not-Rotated 400px", falling back to "Sans Not-Rotated 400px", expect ugly output.      

(process:3860): Pango-WARNING **: 13:30:42.685: couldn't load font "Ubuntu 400px", falling back to "Sans 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:42.690: couldn't load font "Ubuntu Bold Not-Rotated 400px", falling back to "Sans Bold Not-Rotated 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:42.698: couldn't load font "Ubuntu Bold 400px", falling back to "Sans Bold 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:42.701: couldn't load font "Ubuntu Italic Not-Rotated 400px", falling back to "Sans Italic Not-Rotated 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:42.706: couldn't load font "Ubuntu Italic 400px", falling back to "Sans Italic 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:42.708: couldn't load font "Ubuntu Bold Italic Not-Rotated 400px", falling back to "Sans Bold Italic Not-Rotated 400px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:42.712: couldn't load font "Ubuntu Bold Italic 400px", falling back to "Sans Bold Italic 400px", expect ugly output.      

(process:3860): Pango-WARNING **: 13:30:42.718: couldn't load font "Ubuntu Not-Rotated 40px", falling back to "Sans Not-Rotated 40px", expect ugly output.        

(process:3860): Pango-WARNING **: 13:30:42.721: couldn't load font "Ubuntu Bold Not-Rotated 40px", falling back to "Sans Bold Not-Rotated 40px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:42.727: couldn't load font "Ubuntu Italic Not-Rotated 40px", falling back to "Sans Italic Not-Rotated 40px", expect ugly output.

(process:3860): Pango-WARNING **: 13:30:42.730: couldn't load font "Ubuntu Bold Italic Not-Rotated 40px", falling back to "Sans Bold Italic Not-Rotated 40px", expect ugly output.
ok 289 Export clippaths to SVG > ClipPath can be a group with many objects
ok 290 Export clippaths to SVG > Many Objects can share the same clipPath
ok 291 Export clippaths to SVG > an absolute positioned clipPath, shared
ok 292 Export clippaths to SVG > a clipPath on the canvas
ok 293 Export clippaths to SVG > clipPath with a path on a simple elements
ok 294 Export clippaths to SVG > clipPath made of polygons and paths
ok 295 Export clippaths to SVG > Export a radial svg with scaling
ok 296 Export clippaths to SVG > Export complex gradients
ok 297 Export clippaths to SVG > Group with opacity and shadow
ok 298 Export clippaths to SVG > Multiple gradients import
ok 299 Export clippaths to SVG > gradient should be applied to path in svg
1..299
# pass 298
# skip 0
# todo 0
# fail 1
```

**Comment by robothot** [comment]
@ShaMan123  I don't know what caused it, but only this one error. Can you help me to have a look?

File: src/parser/parsePointsAttribute.ts [hunk 1]
```diff
@@ -17,9 +17,9 @@ export function parsePointsAttribute(points) {
...
 
   points = points.split(/\s+/);
-  let parsedPoints = [],
-    i,
-    len;
+  const parsedPoints = [];
+  let i;
+  let len;
 
   for (i = 0, len = points.length; i < len; i += 2) {
```
Comment (ShaMan123) [inline src/parser/parsePointsAttribute.ts:24]:
```suggestion
  for (let i = 0; i < points.length; i += 2) {
```

File: src/parser/selectorMatches.ts [hunk 2]
```diff
@@ -14,10 +12,10 @@ export function selectorMatches(element, selector) {
     matcher = new RegExp('#' + id + '(?![a-zA-Z\\-]+)', 'i');
     selector = selector.replace(matcher, '');
   }
-  if (classNames && selector.length) {
-    classNames = classNames.split(' ');
-    for (i = classNames.length; i--; ) {
-      matcher = new RegExp('\\.' + classNames[i] + '(?![a-zA-Z\\-]+)', 'i');
+  if (classNames != null && selector.length) {
```
Comment (ShaMan123) [inline src/parser/selectorMatches.ts:15]: why did you add this?
I think you should revert this

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+src/mixins/eraser_brush.mixin.ts
+src/util/lang_class.ts
+src/parkinglot/canvas_animation.mixin.ts
```
Comment (ShaMan123) [inline .eslintignore:3]:
```suggestion
src/parkinglot
```

all the folder should be ignored

**Review by ShaMan123 — CHANGES_REQUESTED** [review]
To finish up run
`npm run prettier:write` and add a log to the CHANGELOG.

`build stats` and `coverage` tests fails because your branch is not part of fabric so that is fine. We need to fix that.

File: src/parser/selectorMatches.ts (hunk 2 above)
Comment (robothot) [reply src/parser/selectorMatches.ts:15]: image


This is the method signature `Element.getAttribute(qualifiedName: string): string | null `

So I use the `splitClassNames` variable to store

This method will only have two return values `string` or `null`

Comment (robothot) [reply src/parser/selectorMatches.ts:15]: @ShaMan123  Can you re-examine it for me?  Thank you so much

Comment (ShaMan123) [reply src/parser/selectorMatches.ts:14]: I understand. Good research.
I prefer we leave it as is since `if(null)` will never run and this way we only change lint stuff
Also we use strict equality checks `!==`

Comment (ShaMan123) [reply src/parser/selectorMatches.ts:14]:
```suggestion
  if (classNames && selector.length) {
```

**Comment by ShaMan123** [comment]
> To finish up run `npm run prettier:write` and add a log to the CHANGELOG.
> 

.

File: src/parser/selectorMatches.ts (hunk 2 above)
Comment (robothot) [reply src/parser/selectorMatches.ts]: ok, I understand,

**Comment by robothot** [comment]
@ShaMan123  Do I need to run `npm run prettier:write` and make a commit?

**Comment by ShaMan123** [comment]
yes, exactly

**Comment by robothot** [comment]
@ShaMan123  ok, it's ready

**Review by ShaMan123 — APPROVED** [review]
Great job!
Thanks!
